src/build/arg/mod.rs

```diff
@@ -5109,7 +5114,7 @@ where
                         write(&delim.to_string(), false)?;
                     }
                 }
-                if num_val_names == 1 && mult_val {
+                if (num_val_names == 1 && mult_val) || (arg.is_positional() && mult_occ) {
```

Comment:
I don't see the second branch being tested.

Reply:
Good call, I should have an independent test for this.  This was originally being indirectly covered by subcommands using multiple occurrences but I moved that out of this PR to have that be a separate, explicit decision (like #2977). Our coverage of this shouldn't be tied to the implementation of another feature.